integrations/aws/main.py

```diff
@@ -0,0 +1,196 @@
+    async for session in get_sessions():
+        async for batch in batch_resources(
+            kind,
+            session,
+            "cloudformation",
+            "list_stacks",
+            "StackSummaries",
+            "NextToken",
+        ):
+            yield batch
+
+
+@ocean.on_resync(kind=ResourceKindsWithSpecialHandling.EC2)
+async def resync_ec2(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    async for session in get_sessions():
+        region = session.region_name
+        account_id = await _session_manager.find_account_id_by_session(session)
+        try:
+            async with session.resource("ec2") as ec2:
+                async with session.client("ec2") as ec2_client:
+                    async for page in ec2.instances.pages():
+                        for instance in page:
+                            described_instance = await ec2_client.describe_instances(
+                                InstanceIds=[instance.id]
+                            )
+                            instance_definition = described_instance["Reservations"][0][
+                                "Instances"
+                            ][0]
+                            instance_definition.update(
+                                {
+                                    KIND_PROPERTY: kind,
+                                    ACCOUNT_ID_PROPERTY: account_id,
+                                    REGION_PROPERTY: region,
+                                    IDENTIFIER_PROPERTY: instance.id,
+                                }
+                            )
+                            yield fix_unserializable_date_properties(
+                                instance_definition
+                            )
+        except Exception as e:
+            logger.error(f"Failed to list EC2 Instance in region: {region}; error {e}")
+
+
+@ocean.router.post("/webhook")
+async def webhook(request: Request) -> dict[str, Any]:
+    validate_request(request)
+    try:
+        body = await request.json()
+        logger.info(f"Received AWS Webhook request body: {body}")
+        resource_type = body.get("resource_type")
+        identifier = body.get("identifier")
+        account_id = body.get("accountId")
+        region = body.get("awsRegion")
+
+        with logger.contextualize(
+            account_id=account_id, resource_type=resource_type, identifier=identifier
+        ):
+            if not resource_type or not identifier or not account_id:
+                raise ValueError(
+                    "Resource type, Identifier or Account id was not found in webhook body"
+                )
+
+            matching_resource_configs = get_matching_kinds_from_config(resource_type)
+
+            if not matching_resource_configs:
+                raise ValueError(
+                    "Resource type not found in port app config, update port app config to include the resource type"
+                )
```

Comment:
I want better error handling in order to differentiate between fatal errors to errors which are warning like this one